pkg/acs/calacs/acsccd/blev_funcs_postsm4.c

```diff
@@ -81,102 +117,137 @@ int bias_shift_corr(ACSInfo *acs, SingleGroup *chip2, SingleGroup *chip1) {
   double sum;
   int num;
   double magic_square_mean;
-  
+
   /* allocate space for data arrays */
   ampdata = malloc(arr_rows * arr_cols * sizeof(double));
   ampdata_gap = malloc(nquad_pix * sizeof(double));
   
   /* allocate space for true DC bias levels array */
   dc_bias_levels = malloc((nquad_pix + 1) * sizeof(double));
-  
-  for (i = 0; i < NAMPS; i++) {
-    /* put a single amp's data into ampdata */
-    if (i < 2) {
-      make_amp_array(arr_rows, arr_cols, chip1, i, ampdata);
-    } else {
-      make_amp_array(arr_rows, arr_cols, chip2, i, ampdata);
-    }
+
+  /* For full frame data, there are four amps in use.  For supported subarray
+   * data, there is only one amp in use.  The actual amp in use must be determined. 
+   */
+  int numAmpsInUse = (acs->subarray == YES) ? 1 : NAMPS;
```

Comment:
NAMPS is hard-coded to 4 which nicely accommodates full frame data.  However, the new, "supported" WFC subarrays are single amp (512, 1K, or 2K).   This may have to be updated once the POL and RAMP data are incorporated into this new algorithm.  All other WFC subarrays will use the old blev algorithm.

Reply:
This may have been worth hiding within a func, in case the logic becomes more complicated in the future.